Source/WebInspectorUI/UserInterface/Controllers/JavaScriptLogViewController.js

```diff
@@ -154,22 +156,32 @@ WI.JavaScriptLogViewController = class JavaScriptLogViewController extends WI.Ob
         return consoleMessageView;
     }
 
-    updatePreviousMessageRepeatCount(count, timestamp)
+    updatePreviousMessageRepeatCount(target, count, timestamp)
     {
-        console.assert(this._previousMessageView);
-        if (!this._previousMessageView)
+        let previousMessageView = this._previousMessageView;
+        console.assert(previousMessageView);
+        if (!previousMessageView)
             return false;
 
-        var previousIgnoredCount = this._previousMessageView[WI.JavaScriptLogViewController.IgnoredRepeatCount] || 0;
-        var previousVisibleCount = this._previousMessageView.repeatCount;
+        if (previousMessageView.message.target._identifier !== target._identifier) {
+            previousMessageView = this._lastConsoleMessageViewForTarget.get(target);
+            console.assert(previousMessageView);
+            if (!previousMessageView)
+                return false;
+            this.appendConsoleMessage(previousMessageView.message);
```

Comment:
should we be setting `this._repeatCountWasInterrupted = true;` before adding the new message since the repeat count was actually interrupted (by a message from another target)?

Reply:
Right! Now I fully understood the purpose of this property.